File: lib/types/document.ts
export interface Document {
  id: string;
  filename: string;
  file_type: 'pdf' | 'image' | 'txt' | 'docx' | 'other';
  status: 'pending' | 'processing' | 'ready' | 'failed';
  file_size_bytes: number;
  created_at: string;
  processing_error?: string | null;
}

export interface PaginatedDocumentList {
  items: Document[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

File: backend/app/schemas/document.py
import uuid
from datetime import datetime
from pydantic import BaseModel

class DocumentResponse(BaseModel):
    id: uuid.UUID
    filename: str
    status: str
    file_size_bytes: int
    created_at: datetime

    class Config:
        from_attributes = True

class DocumentDetailResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    filename: str
    file_type: str
    mime_type: str
    file_size_bytes: int
    storage_path: str
    status: str
    processing_error: str | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class PaginatedDocumentResponse(BaseModel):
    items: list[DocumentResponse]
    total: int
    page: int
    page_size: int
    pages: int
